docs/arcadia-users-group/20230228-intro-to-python-1/lesson.md

```diff
@@ -0,0 +1,548 @@
+
+---
+## 1. Variables
+
+Variables in programming languages are containers for assigning values.
+
+Below, we assign the value `1` to the variable `x` and the value `4` to the variable `y`.
+
+
+```python
+x = 1
+y = 4
+```
+
+When we ask Python to evalute `x + y`, it substitutes the value `1` for `x` and the value `4` for `y`.
+
+Running the cell below returns the output value of the expression `x + y`.
+
+
+```python
+x + y
+```
+
+
+
+
+    5
+
+
+
+---
+## 2. Data types
+
+Variables in Python come in a variety of types, such as:
+- `int` or **integer**: whole numbers such as `1`, `2`, `3`, etc.
+- `float` or **floating-point number**: numbers including decimals, e.g. `1.03`.
+- `str` or **string**: chains of alphanumeric characters flanked by quotation marks, e.g. `'apple'`.   
+You can use either single-quotes or double-quotes, but they must be the same on either end.
+
+Python is able to natively perform certain operations using these datatypes, such as basic addition `+`, subtraction `-`, multiplication `*`, division `/`, etc.
+
+
+```python
+1 + 2 * 3.4 / 5
+```
+
+
+
+
+    2.36
+
+
+
+Python also has some helpful intuitive operations – for example, you can join two strings together with a `+` operator as follows:
+
+
+```python
+'pine' + 'apple' 
+```
+
+
+
+
+    'pineapple'
+
+
+
+Certain operations are not supported by default in Python - for example, you can't add a `str` with an `int`. Running the code below will cause a `TypeError`. Python is quite helpful in explaining errors you might encounter. Read the error message below and see if it makes sense to you.
+
+In general, you should expect to encounter *lots* of these errors, or "bugs", when you start getting into programming. **This is totally normal!**  
+In fact, probably 50% of programming is just squashing bugs.  
+
+If you're ever confused about the source of an error, websites such as [StackOverflow](https://stackoverflow.com/) are great places to look for answers.
+
+
+```python
+3 + '5'
+```
+
+
+    ---------------------------------------------------------------------------
+
+    TypeError                                 Traceback (most recent call last)
+
+    Cell In[5], line 1
+    ----> 1 3 + '5'
+
+
+    TypeError: unsupported operand type(s) for +: 'int' and 'str'
+
+
+> **Fun Fact!**
+>
+> The term "bug" predates modern computers, but there are real examples of actual insects interfering with the function of a computer, such as this ["bug" found at Harvard in 1947](https://education.nationalgeographic.org/resource/worlds-first-computer-bug). 
+
+---
+## 3. Using variables
+
+You assign variables in Python using the `variable = value` syntax. You can perform operations on a mix of hard-coded values and variables interchangeably. This is very useful when you expect to use a variable in many different places in your script - Python keeps track of its value so you don't have to.
```

Comment:
```suggestion
You assign variables in Python using the `variable = value` syntax. You can perform operations on a mix of hard-coded values and variables interchangeably. This is very useful when you expect to use a variable in many different places in your script - Python keeps track of its value so you don't have to. If you use variable names that correspond to the value the variable holds (think `wind_mph` for wind speed), this can also help make your code more readable for yourself and others.
```